src/routes/favorites.js

```diff
@@ -1,72 +1,64 @@
 import React from "react";
-import { useHistory } from 'react-router-dom';
 import {
+  Link,
+  useHistory
+} from 'react-router-dom';
+import {
+  FavoritesButton,
   StudySpaceCard,
   Topbar
 } from "../components";
 
-// TODO: Read the JSON file from App.js
-var study_spaces = require("../study-spaces.json");
-var dataHtml = [];
-
-/*
- * Sources of Help:
- *  - Dynamically generating the list of study space cards: https://www.skptricks.com/2018/06/append-or-prepend-html-using-reactjs.html
- **/
-
-const Favorites = () => {
+const Favorites = (props) => {
   const history = useHistory();
 
-  // TODO: Delete this. This is only a temporary solution to the handle click
-  //   callback.
   function handleClick() {
-    console.log("I've been clicked! Bless this day.");
-    // TODO: This is a workaround. Try to use the Link component in the future.
-    // window.location.assign(window.location.href.replace("/favorites", "/view-space"));
+    console.log("Favorites card clicked.");
+    // TODO: This should not be called if we are clicking on the delete button.
     history.push("/view-space");
   }
 
+  /**
+   * Improvements:
+   *  - update list as soon as it's clicked instead of having to refresh the page
+   */
   function handleClickDelete(event) {
-    event.stopPropagation();
-    console.log("I'm being deleted! NOOOO.");
-    // TODO: Use this code when actually deleting from the static db.
-    //var indexToDelete = event.target.getAttribute("index");
-    //dataHtml.splice(indexToDelete, 1);
-    // TODO: This is going to be dummy code for now.
-    document.getElementById("favorites-list").style.display = "none";
+    console.log("Favorites card deleted.");
+
+    // Prevents the Card's onClick/Link from being triggered
+    event.preventDefault();
+
+    // Delete from Local Storage
+    var favoriteSpacesJson = JSON.parse(localStorage.getItem('favoriteSpaces'));
+    var indexToDelete = event.target.getAttribute("index");
+    favoriteSpacesJson.data.splice(indexToDelete, 1);
+    localStorage.setItem('favoriteSpaces', JSON.stringify(favoriteSpacesJson));
   }
-  
-  function renderFavoritesList() {
-    var data = study_spaces.data;
-  
-    dataHtml = [];
-    for (var i = 0; i < data.length; i++) {
-      var space = data[i];
-      dataHtml.push(
+  /* Sources of Help:
+   *  - Interrupting the Link on the Card: https://stackoverflow.com/questions/39849108/disable-react-router-link-in-react
+   **/
+
+  function favoritesListToHtml() {
+    return props.data.data.map((space, i) => {
+      return <Link to={{ state: { data : space }, pathname: "/view-space" }} style={{ textDecoration: "none", color: "#000000" }}>
```

Comment:
Can you double check to make sure this works? Worried about conflicts between link and the onClick function you have. Seems a little redundant, but if its working then i guess thats fine(?)

Reply:
removed onClick - link works